google/cloud/storage/list_buckets_reader_test.cc

```diff
@@ -0,0 +1,97 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storage/list_buckets_reader.h"
+#include "google/cloud/storage/internal/nljson.h"
+#include "google/cloud/storage/testing/canonical_errors.h"
+#include "google/cloud/storage/testing/mock_client.h"
+#include <gmock/gmock.h>
+
+namespace google {
+namespace cloud {
+namespace storage {
+inline namespace STORAGE_CLIENT_NS {
+using internal::ListBucketsRequest;
```

Comment:
This looks backwards; the two styles I've seen for using-directives-in-cc-files are:

1) Add using-directives (and test-only structures) in an unnamed namespace, and everything else in the enclosing namespace
2) Add everything in an unnamed namespace (I prefer this one, personally).

It looks like you've done the reverse of option #1 :)

Reply:
Fixed.